src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1284,18 +1284,25 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
                     null, Metadata.ValidateCountNotInArray);
             }
 
+            if (MinLength == MaxLength && len != MaxLength)
+            {
+                throw new ValidationMetadataException("ValidateCountEqualLengthFailure",
```

Comment:
The error id `ValidateCountEqualLengthFailure` sounds confusing. Maybe `ValidateCountNotExactlyEqual`?